### ISSUE TITLE
[SourceSearch] Search Bar results disappears when resizing Firefox's window

### ISSUE THREAD

**Issue by jasonLaster** [devtools-html/debugger.html#3714]
[Steps to reproduce]:
1.Launch Firefox.
2.Access the https://devtools-html.github.io/debugger-examples/examples/todomvc/ webpage.
3.Open the debugger.
4.Press CTR+P.
5.Type something in the search bar.
6.Resize Firefox's window.

[Expected result]:
Search results are displayed.

[Actual result]:
Search results disappear after resizing Firefox's window.

![Resizing the search panel.gif](https://shipusercontent.com/d5fda0345a567c8b1f88d75b6070929e/Resizing%20the%20search%20panel.gif)

**Comment by rohanprasad** [comment]
@jasonLaster I was trying to debug this and found that the query is not added to the store which is causing the error. In the autocomplete component the onChange event sets the query string in state of autocomplete. But on resizing, due to SplitBox the components are destroyed and re-rendered, which causes the query string to be lost.

Working on a fix.

**Comment by bomsy** [comment]
@rohanprasad nice! will mark this in progress

**Comment by rohanprasad** [comment]
So I did some quick tweaks and got things working. But currently I used https://github.com/devtools-html/debugger.html/blob/master/src/reducers/project-text-search.js to save the result. I'll create a similar reducer for source-search (correct me if I am wrong).

I was also working on https://github.com/devtools-html/debugger.html/blob/master/src/components/shared/Autocomplete.js and would like to make changes to convert state variables to props (to save them in store and use from there). Need feedback for this.
